### DISCUSSION TITLE
Is PutQuorum set too conservatively in CodeMode?

### DISCUSSION THREAD

**Discussion by FourSpaces** [cubefs/cubefs#3486] [General]
Is PutQuorum set too conservatively in CodeMode?
Is PutQuorum set too conservatively in CodeMode, which will result in the inability to continue writing after losing nodes. Although the code_rode_put_quorum parameter can be set to PutQuorum, the setting is too cumbersome. To simplify this setting, it is best to set it in the code_rode_policys parameter of Clustermgr

**Comment by JasonHu520** [comment]
In order to ensure the reliability of data, the Quorum mechanism needs to ensure that the data that has been written has a certain data self-recovery capability. When the cluster writes cannot guarantee Quorum, the cluster is already in an unhealthy state, and the risk of continuing to write at this time is also higher,Therefore, it is generally not recommended to adjust Quorum.

**Reply by FourSpaces** [reply]
@JasonHu520 In erasure codes, even if N data blocks, M check blocks, and M fragments are lost, data can still be recovered. In the code, the configuration value of PutQuorum is usually N+M+L - AZCount, which I think can be bolder. For example, in single AZ scenarios such as EC12P4, EC16P4, EC3P3, EC10P4, and EC6P3, losing two shards is normal and recoverable

**Reply by JasonHu520** [reply]
consider the disk repair scenario, there are many bids on a disk, and the combination of successful writes on the strip for each bid is different. The quorum mechanism is mainly used to deal with disk damage via data has been written. Anti-risk ability, Since data repair is asynchronous, disk damage during this period has a great impact on data security. If we only consider the reduction of quorum value under normal circumstances, it may not have an impact.
